src/backend/distributed/metadata/metadata_utility.c

```diff
@@ -4419,3 +4437,17 @@ UnblockDependingBackgroundTasks(BackgroundTask *task)
 
 	table_close(pgDistBackgroundTasksDepend, NoLock);
 }
+
+/*
+ * AutoConverted attr was added to table pg_dist_partition using alter operation, so
+ * in case of downgrade we can get additional tuple field in tuple desc with dropped flag
+ * and in case of futher upgrade we can get number of fields > Natts_pg_dist_partition.
+ * So we should properly address AutoConverted attr in arrays.
+ */
```

Comment:
We usually start function comments with the name of the function, also proposing slight improvements for the comment:

```suggestion
/*
 * GetAutoConvertedAttrIndexInPgDistPartition returns attrnum for autoconverted attr.
 *
 * autoconverted attr was added to table pg_dist_partition using alter operation after
 * the version where Citus started supporting downgrades, and it's only column that we've
 * introduced to pg_dist_partition since then.
 *
 * And in case of a downgrade + upgrade, tupleDEsc->natts becomes greater than
 * Natts_pg_dist_partition and when this happens, then we know that attrnum autoconverted is
 * not Anum_pg_dist_partition_autoconverted anymore but tupleDEsc->natts - 1.
 */
```